rasa/nlu/test.py

```diff
@@ -1319,94 +1308,34 @@ def get_eval_data(
                     example.get(ENTITIES, []),
                     result.get(ENTITIES, []),
                     result.get(TOKENS_NAMES[TEXT], []),
-                    result.get(TEXT, ""),
+                    result.get(TEXT),
                 )
             )
 
     return intent_results, response_selection_results, entity_results
 
 
-def get_entity_extractors(interpreter: Interpreter) -> Set[Text]:
-    """Finds the names of entity extractors used by the interpreter.
-
-    Processors are removed since they do not detect the boundaries themselves.
-
-    Args:
-        interpreter: the interpreter
-
-    Returns: entity extractor names
-    """
-    from rasa.nlu.extractors.extractor import EntityExtractor
-    from rasa.nlu.classifiers.diet_classifier import DIETClassifier
-
-    extractors = set()
-    for c in interpreter.pipeline:
-        if isinstance(c, EntityExtractor):
-            if isinstance(c, DIETClassifier):
-                if c.component_config[ENTITY_RECOGNITION]:
-                    extractors.add(c.name)
-            else:
-                extractors.add(c.name)
-
-    return extractors - ENTITY_PROCESSORS
-
-
-def is_entity_extractor_present(interpreter: Interpreter) -> bool:
-    """Checks whether entity extractor is present."""
-    extractors = get_entity_extractors(interpreter)
-    return len(extractors) > 0
-
-
-def is_intent_classifier_present(interpreter: Interpreter) -> bool:
-    """Checks whether intent classifier is present."""
-    from rasa.nlu.classifiers.classifier import IntentClassifier
-
-    intent_classifiers = [
-        c.name for c in interpreter.pipeline if isinstance(c, IntentClassifier)
-    ]
-    return intent_classifiers != []
-
-
-def is_response_selector_present(interpreter: Interpreter) -> bool:
-    """Checks whether response selector is present."""
-
-    from rasa.nlu.selectors.response_selector import ResponseSelector
-
-    response_selectors = [
-        c.name for c in interpreter.pipeline if isinstance(c, ResponseSelector)
-    ]
-    return response_selectors != []
-
-
-def get_available_response_selector_types(
-    interpreter: Interpreter,
-) -> List[Optional[Text]]:
-    """Gets all available response selector types."""
-    from rasa.nlu.selectors.response_selector import ResponseSelector
-
-    response_selector_types: List[Optional[Text]] = [
-        c.retrieval_intent
-        for c in interpreter.pipeline
-        if isinstance(c, ResponseSelector)
-    ]
-
-    return response_selector_types
-
-
-def remove_pretrained_extractors(pipeline: List[Component]) -> List[Component]:
-    """Remove pre-trained extractors from the pipeline.
-
-    Remove pre-trained extractors so that entities from pre-trained extractors
-    are not predicted upon parsing.
+def get_active_entity_extractors(
```

Comment:
This is the root of all problems, right? What do you think of changing the predictions of entity extractors to do return `{"name": "<name of entity extractor", "entities": [...]}`? We could keep the change only with NLU for now and keep the `UserUttered` part the same.

Reply:
Yeah that makes sense! Then you would know reliably what extractors are active in the graph 💡

Reply:
@dakshvar22 What do you think of this suggestion?

Reply:
can this be protected?
```suggestion
def _get_active_entity_extractors(
```

Reply:
@twerkmeister I think it's fine to keep it like this for now but can we please create a follow up issue? Then we can either change the entity predictions or allow some basic introspection (e.g. the model metadata has the graph schema which would allow us iterating over classes from the outside)

Reply:
thanks!